pytransform3d/transform_manager/_temporal_transform_manager.py

```diff
@@ -50,6 +54,86 @@ def as_matrix(self, time):
     def check_transforms(self):
         self._A2B = check_transform(self._A2B)
         return self
+    
+
+class PandasTimeseriesTransform(TimeVaryingTransform):
+    """Transformation, which does change over time, represented in a pandas
+    dataframe.
+
+    Parameters
+    ----------
+    df : dataframe, shape (N, 7+)
+        Time-sequence of transformations, with each row representing a single
+        sample as position-quarternion (PQ) structure.
+        Refer to `pq_column_names()` for column naming.
+
+    """
+    def __init__(self, df):
+        self._df = df
+
+    @property
+    def _time_index(self):
+        return self._df.index.to_numpy()
+    
+    def _check_if_time_is_in_bounds(self, time):
+        time_index = self._time_index
+
+        if (time < time_index[0]) or (time > time_index[-1]):
+            raise ValueError("Time value is out of range!")
+
+    def as_matrix(self, time):
+        self._check_if_time_is_in_bounds(time)
+        pq = self._interpolate_on_demand(time)
+        return transform_from_pq(pq)
+    
+    def _interpolate_on_demand(self, time):
+        # interpolate only if needed, otherwise take the corresponding
+        # sample from the dataframe
+        indices = find_neighboring_indices_around_value(self._time_index, time)
+        if indices[0] == indices[1]:
+            # Match, no interpolation needed, we just take the sample!
+            pq = self.pq_from_record(indices[0])
+        else:
+            pq = self._interpolate_pq(time)
+        return pq
+
+    def _interpolate_pq(self, time):
+        # identify the index of the preceding sample
+        t_arr = self._time_index
+        idx_timestep_earlier_wrt_query_time = np.argmax(t_arr >= time) - 1
+
+        # TODO: maybe not that efficient to do this
+        pq_array = self._df[self.pq_column_names()].to_numpy()
```

Comment:
If we copy this to a numpy array anyway, do we really still need pandas for this? We could just do .to_numpy() when we create the dynamic transformation object and use numpy arrays internally. I don't really see the benefit of using pandas at the moment. Then it would be a bad idea to restrict the interface to pandas as it adds another dependency.

Maybe it is also worth the effort to take a look at [sklearn-pandas's](https://github.com/scikit-learn-contrib/sklearn-pandas) approach. They have a similar problem: sklearn works with numpy arrays internally and they provide a pandas interface.

Reply:
Hm, we can just deal with (N,) time array and (N,7) PQ array to avoid pandas completely? Currently you are right, pandas is not needed ...

Reply:
That is currently the best solution in my opinion. If needed, it should not be too hard to implement the pandas version based on the numpy version.

Reply:
Adapted to two numpy arrays.